lib/rewrite/rewrite-expr-parser.c

```diff
@@ -36,6 +36,7 @@ static CfgLexerKeyword rewrite_expr_keywords[] =
   { "subst",              KW_SUBST },
   { "set_tag",            KW_SET_TAG },
   { "clear_tag",          KW_CLEAR_TAG },
+  { "set_severity",       KW_SET_SEVERITY },
```

Comment:
I am considering adding `set_level` and `set_priority` to this list as alternatives to `set_severity`: while the RFC use *severity*, the syslog-ng doc use *level* and *priority* a lot, and those are the words I was using when I found the original PR.  What do you think?

Reply:
the problem is that "priority" in RFC parlance is the combination of "severity" and "facility". The syslog implementation in libc used the term "priority" as a combination of "facility" and "level":

>        The priority argument is formed by ORing together a facility value and a level value (described below).  If no facility value is ORed into priority, then the default value set by openlog() is used, or, if there was no preceding openlog() call, a default of LOG_USER is employed.

Unfortunately, syslog-ng was not very consistent with this naming, for instance we have a filter expression called priority() that is the same as level().

I would try to push for more consistency.

So I would opt for using the term "severity" with "level" as an obsolete alias, any inconsistent use should be obsoleted too (like the priority() filter expression). And if level is obsolete, I wouldn't add set-level().

I can be convinced that this is too pushy, in that case we should be adding level() aliases everywhere and not complain about them at all. (e.g. don't obsolete that syntax either). The reason I would be ok with this idea is that a _lot_ of users already have that syntax in their configuration and changing those config files now is a lot of work we would be pushing to the world. An alias here and there is not a maintenance nightmare. If this is what we decide, we should indeed add set-level(), but not set-priority().

My $.02

Reply:
Thanks for the details, I understand better why those words have blur meanings in my head :smile:!

As far as I am concerned, I am in favor of improving consistency and deprecating the confusing words, therefore not adding aliases.  But I guess this is a decision that is up to the project so if you decide to go the other route, I'll add the aliases.

Reply:
@pzoleex can you help us decide here, please?